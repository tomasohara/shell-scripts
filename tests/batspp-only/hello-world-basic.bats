#!/usr/bin/env bats
#
# This test file was generated using Batspp
# https://github.com/LimaBD/batspp
#

# Constants
VERBOSE_DEBUG=""
TEMP_DIR="/tmp/batspp-106732"

# Setup function
# $1 -> test name
function run_setup () {
	test_folder=$(echo $TEMP_DIR/$1-$$)
	mkdir --parents "$test_folder"
	cd "$test_folder" || echo Warning: Unable to "cd $test_folder"
	bind 'set enable-bracketed-paste off'
	echo "BIND ON!"
}

# Teardown function
function run_teardown () {
	: # Nothing here...
}

@test "test of line 5" {
	run_setup "test-of-line-5"

	# Assertion of line 5
	shopt -s expand_aliases
	print_debug "$(echo "Hi Mom!")" "$(echo -e 'Hi Mom!\n')"
	[ "$(echo "Hi Mom!")" == "$(echo -e 'Hi Mom!\n')" ]

	run_teardown
}

@test "test of line 8" {
	run_setup "test-of-line-8"

	# Assertion of line 8
	shopt -s expand_aliases
	print_debug "$(echo 'Hello world' | wc -l)" "$(echo -e '1\n')"
	[ "$(echo 'Hello world' | wc -l)" == "$(echo -e '1\n')" ]

	run_teardown
}

# This prints debug data when an assertion fail
# $1 -> actual value
# $2 -> expected value
function print_debug() {
	echo "=======  actual  ======="
	bash -c "echo \"$1\" $VERBOSE_DEBUG"
	echo "======= expected ======="
	bash -c "echo \"$2\" $VERBOSE_DEBUG"
	echo "========================"
}
