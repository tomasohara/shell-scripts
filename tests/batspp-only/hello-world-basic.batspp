# Setup
$ bind 'set enable-bracketed-paste off'
$ echo "BIND ON!"

$ echo "Hi Mom!"
Hi Mom!

$ echo 'Hello world' | wc -l
1
