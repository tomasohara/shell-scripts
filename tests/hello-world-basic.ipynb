{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a03c1f14",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# Global Setup\n",
    "bind 'set enable-bracketed-paste off'\n",
    "## OLD: echo 'BIND ON!' returns assertion failed (unexpected header file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "619ceccf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BIND ON!\n"
     ]
    }
   ],
   "source": [
    "## This is how a simple command works\n",
    "echo 'BIND ON!'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "93dfcf88",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n"
     ]
    }
   ],
   "source": [
    "## Here's an example of pipeline\n",
    "echo 'Hello world' | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "adb5fa38",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n"
     ]
    }
   ],
   "source": [
    "## Tests using single line conditions and echo\n",
    "[ \"$(echo 'Hi Mom!' | wc -c)\" -eq 8 ]; echo $? "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
