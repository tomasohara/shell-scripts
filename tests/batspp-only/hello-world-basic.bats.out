1..3
not ok 1 test-1
# (in test file batspp-only/hello-world-basic.bats, line 21)
#   `[ "$(test-1-actual)" == "$(test-1-expected)" ]' failed
# ========== bind 'set enable-bracketed-paste off'\n ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-only/hello-world-basic.bats: line 28: bind: warning: line editing not enabled
# ========= $ echo "BIND ON!" =========
# $ echo "BIND ON!"
# ============================
# /home/aveey/tom-project/shell-scripts/tests/batspp-only/hello-world-basic.bats: line 28: bind: warning: line editing not enabled
ok 2 test-3
ok 3 test-4
